tron/eventbus.py

```diff
@@ -88,20 +91,28 @@ def sync_save_log(self, reason):
         log.info(f"log dumped to disk because {reason}, took {duration:.4}s")
 
     def sync_loop(self):
-        if self.must_shutdown:
-            self.sync_save_log("shutdown")
-            log.info("shutdown completed")
-        else:
+        if not self.enabled:
+            return
+
+        try:
             self.sync_process()
-            reactor.callLater(1, self.sync_loop)
+        except Exception as e:
+            log.error(e)
```

Comment:
think this could be dangerous? I'd hate for us to not notice these flying by - maybe we should exit if this fails?

Reply:
i was considering crashing the process here but decided to just log the error because no essential functionality is affected, but it's not a strong opinion, i'll change it.